core/src/operations/decorators/advice.rs

```diff
@@ -3,37 +3,38 @@ use core::fmt;
 /// TODO: add docs
 #[derive(Copy, Clone, Debug, Eq, PartialEq)]
 pub enum AdviceInjector {
-    /// Injects a node of the Merkle tree specified by the values on the stack at the head of the
-    /// advice tape. The stack is expected to be arranged as follows (from the top):
+    /// Pushes a node of the Merkle tree specified by the values on the top of the operand stack
+    /// onto the advice stack. The operand stack is expected to be arranged as follows (from the
+    /// top):
     /// - depth of the node, 1 element
     /// - index of the node, 1 element
     /// - root of the tree, 4 elements
     MerkleNode,
 
-    /// Injects the result of u64 division (both the quotient and the remainder) at the head of
-    /// the advice tape. The stack is expected to be arranged as follows (from the top):
+    /// Pushes the result of [u64] division (both the quotient and the remainder) onto the advice
+    /// stack. The operand stack is expected to be arranged as follows (from the top):
     /// - divisor split into two 32-bit elements
     /// - dividend split into two 32-bit elements
     ///
-    /// The result is injected into the advice tape as follows: first the remainder is injected,
-    /// then the quotient is injected.
+    /// The result is pushed onto the advice stack as follows: first the remainder is pushed,
+    /// then the quotient.
     DivResultU64,
 
-    /// Injects a list of field elements at the front of the advice tape. The list is looked up in
-    /// the key-value map maintained by the advice provider using the top 4 elements on the stack
-    /// as the key.
+    /// Pushes a list of field elements onto the advice stack. The list is looked up in the
+    /// key-value map maintained by the advice provider using the top 4 elements of the operand
+    /// stack as key.
     MapValue,
 
-    /// Injects a list of words from the memory starting from the specified start address.
+    /// Pushes a list of words from the memory starting from the specified address.
```

Comment:
This actually does not affect the advice stack and works only with the advice map. The original description here could be improved as follows:

```Rust
/// Inserts a list of words read from VM memory into the advice map under the key specified by
/// the word at the top of the operand stack.
///
/// The first internal u32 specifies the starting address of the memory region, and the second
/// u32 specifies the number of words to be read.
```

Reply:
Fixed on https://github.com/0xPolygonMiden/miden-vm/pull/785

Reply:
Fixed on #785